src/liballoc/boxed.rs

```diff
@@ -623,6 +624,79 @@ impl From<Box<str>> for Box<[u8]> {
     }
 }
 
+/// The error type returned when a conversion from a boxed slice to a boxed
+/// array fails.
+#[unstable(feature = "try_from", issue = "33417")]
+#[derive(Clone)]
+pub struct TryFromSliceError<T>(Box<[T]>);
```

Comment:
Please add a brief example involving TryFromSliceError. See [FromUtf8Error](https://doc.rust-lang.org/std/string/struct.FromUtf8Error.html) for examples of examples.